File: app/src/main/java/com/prey/PreyVerify.java
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2015 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey;

import com.prey.net.PreyWebServices;

import android.content.Context;

public class PreyVerify {

    private static PreyVerify instance = null;

    private PreyVerify(Context ctx) {
        init(ctx);
    }

    public static PreyVerify getInstance(Context ctx) {
        if (instance == null) {
            instance = new PreyVerify(ctx);
        }
        return instance;
    }

    private void init(Context ctx) {
        final Context myContext = ctx;
        new Thread() {
            public void run() {
                try {
                    sleep(2000);
                } catch (InterruptedException e) {
                }
                try {
                    PreyWebServices.getInstance().verify(myContext);
                } catch (Exception e) {
                }
            }
        }.start();
    }

}

File: app/src/main/java/com/prey/PreyVerify.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey

/**
 * Represents a Prey verification result, containing a status code and description.
 */
class PreyVerify {

    private var statusCode: Int = -1
    private var statusDescription: String? = null
    fun getStatusCode(): Int {
        return statusCode
    }

    fun setStatusCode(statusCode: Int) {
        this.statusCode = statusCode
    }

    fun getStatusDescription(): String {
        return statusDescription!!
    }

    fun setStatusDescription(statusDescription: String?) {
        this.statusDescription = statusDescription
    }

}